lib/Doctrine/DBAL/SQLParserUtils.php

```diff
@@ -98,7 +103,7 @@ static public function expandListParameters($query, $params, $types)
         foreach ($types as $name => $type) {
             ++$bindIndex;
 
-            if ($type !== Connection::PARAM_INT_ARRAY && $type !== Connection::PARAM_STR_ARRAY) {
+            if (!in_array($type, $relevantArrayTypes)) {
```

Comment:
Instead of making this `if` extra long and checking 3 different times, I placed the types into an array and am doing a `!in_array()` call